airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py

```diff
@@ -158,21 +165,37 @@ def _raise_error_listing_files(self, globs: List[str], exc: Optional[Exception]
             endpoint=self.config.endpoint,
         ) from exc
 
-    def open_file(self, file: RemoteFile, mode: FileReadMode, encoding: Optional[str], logger: logging.Logger) -> IOBase:
+    def open_file(
+        self,
+        file: RemoteFile,
+        mode: FileReadMode,
+        encoding: Optional[str],
+        logger: logging.Logger,
+    ) -> IOBase:
         try:
             params = {"client": self.s3_client}
         except Exception as exc:
             raise exc
 
+        fully_qualified_uri = self.get_fully_qualified_uri(file.uri)
         logger.debug(f"try to open {file.uri}")
         try:
             if isinstance(file, RemoteFileInsideArchive):
-                s3_file_object = smart_open.open(f"s3://{self.config.bucket}/{file.uri.split('#')[0]}", transport_params=params, mode="rb")
+                s3_file_object = smart_open.open(
+                    fully_qualified_uri,
+                    transport_params=params,
+                    mode="rb",
+                    buffering=DEFAULT_BUFFER_SIZE,
```

Comment:
Added explicit buffering param (40MB per file) to target higher throughput. I couldn't find reference to the exact default buffer size, but I think it was 5MB or 10MB if not otherwise specified.